### DISCUSSION TITLE
lfric_apps rose stem --group=build not working?

### DISCUSSION THREAD

**Discussion by MichaelWhitall** [MetOffice/simulation-systems#279] [LFRic]
I folks,

I'm working on some linked UM / lfric_apps tickets:
https://code.metoffice.gov.uk/trac/um/ticket/7815
https://code.metoffice.gov.uk/trac/lfric_apps/ticket/467

and just wanted to test the build from the lfric_apps side (for a range of compilers) before updating all the rose stem app namelists etc.  I found in the working practices:
https://metoffice.github.io/simulation-systems/WorkingPractices/TestSuites/lfric_apps.html
there's mention of a "build" group in the lfric_apps rose stem, which sounds like what I need :)  So I tried running rose stem from my lfric_apps branch with --group=build.

However, this seems to've only run the first task in the tree (remote-init_spice) and then just stalled without submitting anything else.

I tried manually triggering some of the other tasks that were just sitting there waiting and not submitting, but this resulted in numerous things failing with "No such file or directory" errors.

Anyone know if I can run the "build" rose stem group / what I need to do differently?

Cheers!
Mike

**Comment by MichaelWhitall** [comment]
Update:  for some reason I found I needed to manually trigger the "remote-init_xc40" and "export source" apps (at-which point numerous compiles then auto-trigger and fail due to "No such file or directory"), but then manually re-trigger the fcm_make tasks again after the remote-init and export source tasks have finished  After doing that, the rest of the tree seems to've sprung into life and my test lfric_apps compiles are running :)

But something seems to be wrong with the graph for lfric_apps rose stem if this much manual intervention is required?

**Comment by james-bruten-mo** [comment]
Hi Mike, thanks for pointing that out. You're right, there is an issue with the graphing for the xc40 build group caused by the lfric_coupled fcm_make_drivers task not having a dependency (specific of course to build only tasks). The quickest fix if you want to run the build group is to remove the coupled build - comment out line 44 in rose-stem/site/meto/groups/groups_lfric_coupled.cylc.

**Reply by MichaelWhitall** [reply]
No problem, awesome thanks James I'll try and remember that for next time...
I'm now through to testing the apps at run-time so using --group=developer for now :)

Cheers!
Mike
